compaction/compaction_manager.cc

```diff
@@ -1034,8 +1034,14 @@ void compaction_manager::register_metrics() {
 }
 
 void compaction_manager::enable() {
-    SCYLLA_ASSERT(_state == state::none || _state == state::disabled);
-    _state = state::enabled;
+    SCYLLA_ASSERT(_state == state::none || _state == state::running);
+
+    if (_disabled_state_count > 0 && --_disabled_state_count > 0) {
+        cmlog.debug("Compaction manager is still disabled, requires {} more call(s) to enable()", _disabled_state_count);
+        return;
+    }
+
+    _state = state::running;
```

Comment:
I'd consider the following so that `_disabled_state_count` can be initialized to 0:
```c++
if (_state != state::running) {
    _state = state::running;
} else if (_disabled_state_count > 0 && --_disabled_state_count > 0) {
    cmlog.debug("Compaction manager is still disabled, requires {} more call(s) to enable()", _disabled_state_count);
    return;
}
```